fastfield_codecs/src/linear.rs

```diff
@@ -126,18 +126,21 @@ impl FastFieldCodec for LinearCodec {
             return None; // disable compressor for this case
         }
 
-        // let's sample at 0%, 5%, 10% .. 95%, 100%
-        let num_vals = column.num_vals() as f32 / 100.0;
-        let sample_positions = (0..20)
-            .map(|pos| (num_vals * pos as f32 * 5.0) as u64)
-            .collect::<Vec<_>>();
+        let limit_num_vals = column.num_vals().min(100_000);
+
+        let num_samples = 100;
```

Comment:
I like the step_by solution. 

Can you change the column.num_vals() limit above to 100 too, to make proofreading even easier.

Reply:
I don't understand, 100 instead of 100_000?